lib/msf/core/handler/reverse_http.rb

```diff
@@ -89,13 +91,23 @@ def listener_uri
   #
   # @return [String] A URI of the form +scheme://host:port/+
   def payload_uri(req)
-    if req and req.headers and req.headers['Host'] and not datastore['OverrideRequestHost']
+    callback_host = nil
+
+    # Extract whatever the client sent us in the Host header
+    if req and req.headers and req.headers['Host']
```

Comment:
Legend has it that use ```&&``` instead of ```and``` is better for the happy ending. :bride_with_veil::two_hearts::man: 
=P